### ISSUE TITLE
Add toggle for --hold in konsole

### ISSUE THREAD

**Issue by JackLeEmmerdeur** [dhruv8sh/arch-update-checker#6]
There was a single package shown in the update-list.

1. Clicked install on that package in the update list
2. Install process opens in a Konsole window
3. Hangs on the last step e.g. `(4/4) Updating the desktop file MIME type cache...`
4. Process in konsole can't be closed with CTRL+C only the whole window with ALT+F4
5. The applet still shows available updates

A regular `yay` or `paru` in an external Konsole window runs without problems.
Also the "Update your system" in the applet runs successfully and closes the Konsole window.

**Comment by dhruv8sh** [comment]
Your package probably finished installing.
For testing purposes I enabled the ```--hold``` flag for konsole so it does not exit right after installing.
Some people might want this so I will add a toggle for this in the next release.

Although, if you really really want to fix this right now,
Do
``` nano ~/.local/share/plasma/plasmoids/org.kde.archupdatecheker/contents/ui```
and simply remove ```--hold``` from the strings in the function ```installOnly```.
Save and exit
Run ```systemctl --user restart plasma-plasmashell```

**Comment by JackLeEmmerdeur** [comment]
Ok understood. 

> Do
> nano ~/.local/share/plasma/plasmoids/org.kde.archupdatecheker/contents/ui
> and simply remove --hold from the strings in the function installOnly.

You got a typo in `archupdatecheker` and `ui` is a directory.
You meant `--hold` in the installOnly() function in PackageManager.qml, right?

**Comment by dhruv8sh** [comment]
Yes, my bad. I am using my phone to type this. Did not review what I wrote.

**Comment by JackLeEmmerdeur** [comment]
All good.
Just wanted to point at possible confusion that could come up from others watching this thread.
What was irritating about this is the --hold option would definitely prevent the installation of the single package as if the process was stopped at the very last moment.

I think I'll keep the --hold option to prevent myself from updating singular packages as I usually don't want to do partial upgrades.

**Comment by dhruv8sh** [comment]
Good point. I will definitely hide the installation of single packages from AUR/Pacman behind an 'Use at your own risk' flag.

**Comment by JackLeEmmerdeur** [comment]
> Good point. I will definitely hide the installation of single packages from AUR/Pacman behind an 'Use at your own risk' flag.

Yes do it. Or feel sorry later when lot of people with broken systems will congregate here in the issues later.

But I still can report that the konsole window doesn't reach the point where the package is really installed.
When the install process holds I checked the output of `pacman -Syu` and the package was still in the list.
Also your applet shows it still.
I can only quit the konsole window via ALT+F4.
And then the package definitely isn't updated.
So you shouldn't have to fear broken systems currently when you keep the "--hold" flags in the code.

**Comment by dhruv8sh** [comment]
It definitely does install the package for me. Can you re-test and confirm?

```--hold``` command makes it so that ```Ctrl+C``` cannot exit after the installation, because there is no process running to interrupt. It will work when the package is being installed.

![image](https://github.com/dhruv8sh/arch-update-checker/assets/67322047/c38fd0c0-d40c-46bf-9995-abf979bf774c)
```Ctrl+C``` when the process is running.

The flag for this has been added anyways and should be pushed within an hour alongside some minor bug-fixes.

**Comment by JackLeEmmerdeur** [comment]
Thanks for the clarification about the process handling.

> Can you re-test and confirm?

It was definitely like that at the time of my last post. I tested it a few times.
Maybe it was a system or pacman setting that interfered.

Anyways I tested the new version and the the installation of individual packages now works.

Nice UX regarding the alert (and the subsequent greying out of the install button).
The alert-message is very non-ambiguous.
Nobody can say they haven't been warned.

Kudos to your work and TYVM. This applet is minimalistic and stylish and seems to work really good now.
Exactly what beginners need as [discover's update notifier](https://bugs.kde.org/show_bug.cgi?id=466248) seems to be buggy.
